File: Protótipos(back-end)/projetoFila(java)/Funcionario.java
package projetoMascate;

import java.util.Map;

public class Funcionario {
    private String nome;
    private String cargo;
    private FilaDisplay filaDisplay;
    
    public Funcionario() {}; 

    public Funcionario(String nome, String cargo, FilaDisplay filaDisplay) {
        this.nome = nome;
        this.cargo = cargo;
        this.filaDisplay = filaDisplay;
    }

    public String getNome() {
        return nome;
    }

    public String getCargo() {
        return cargo;
    }

    public void visualizarFila(Map<Integer, Fila> filasEspera) {
        filaDisplay.exibirFilas(filasEspera);
    }

    public void visualizarClienteDaVez(Map<Integer, Fila> filasEspera) {
        for (Map.Entry<Integer, Fila> entry : filasEspera.entrySet()) {
            Fila fila = entry.getValue();
            if (!fila.getClientes().isEmpty()) {
                Cliente cliente = fila.getClientes().get(0);
                System.out.println("Cliente da vez para fila de " + fila.getLugares() + " lugares:");
                System.out.println("Nome: " + cliente.getNome() + ", Senha: " + String.format("%04d", cliente.getSenha()));
                return;
            }
        }
        System.out.println("Não há clientes na fila.");
    }
}

File: Protótipos(back-end)/projetoFila(ts)/Funcionario.ts
import Fila from "./Fila";
import FilaDisplay from "./FilaDisplay";

class Funcionario {
    private nome: string;
    private cargo: string;
    private filaDisplay: FilaDisplay;

    constructor(nome: string, cargo: string, filaDisplay: FilaDisplay) {
        this.nome = nome;
        this.cargo = cargo;
        this.filaDisplay = filaDisplay;
    }

    getNome(): string {
        return this.nome;
    }

    getCargo(): string {
        return this.cargo;
    }

    visualizarFila(filasEspera: Map<number, Fila>): void {
        this.filaDisplay.exibirFilas(filasEspera);
    }

    visualizarClienteDaVez(filasEspera: Map<number, Fila>): void {
        for (const [_, fila] of filasEspera) {
            const clientes = fila.getClientes();
            if (clientes.length > 0) {
                const cliente = clientes[0];
                console.log(`Cliente da vez para fila de ${fila.getLugares()} lugares:`);
                console.log(`Nome: ${cliente.getNome()}, Senha: ${cliente.getSenha().toString().padStart(4, '0')}`);
                return;
            }
        }
        console.log("Não há clientes na fila.");
    }
}

export default Funcionario;
